### PR TITLE
Dynamic now plays a part in controlling random event antagonists

### PR DESCRIPTION
**Pull request by Mothblocks**
## About The Pull Request
image

`n` is currently 10 to 18 minutes.

---> **Marked as DNM, as I want playtesting to see if this is the best idea. If a maintainer is interested in merging this, let me know and I can perform playtesting starting with Manuel (both for bugs and for fun).** <---

### ...turn off random event antags entirely?

This would make greenshifts completely barren of antagonists, which I don't believe is a positive. Greenshifts are still intended to have random antags, as seen by the extended mode not disabling them either. Checking for threat level is also not an appropriate solution, as low-ish threat rounds (think 30) can become effectively greenshifts if all the antags suck.

### ...generate threat?

Generating more threat from the round is something I want to do, but doing it based on time is something I'm wary of, as it would necessarily force a rethink in the way the entire midround injection system works. A lot of the easy ways to generate threat (such as antagonists dying) are not something that works on greenshifts. Basically, it's not that I'm confident generating threat is a *worse* move to take, but it's a pretty large design question that I can't solidly answer yet.

### ...make random event antags add threat instead?

I'm still thinking about it. My most confident answer is that I don't want Dynamic to see active, well-paced shifts and to think that it needs to inject even more threat, and potentially spawn something too catastrophic like Nuclear Assault. The solution outlined in the flowchart keeps random event antags on Dynamic's pace, but still has the possibility to go out of sync if a midround injection is not coming soon.

## Why It's Good For The Game
Dynamic 2021's carefully tuned pacing is demolished by random event antags spawning new antags right after Dynamic spawns a midround. While I do in some respect believe this is simply the roll of the dice, it's just too volatile for me to feel justified ignoring it.

## NOT in this PR

The ability for admins to cancel midround events. This is coming soon, I promise.

## Changelog
:cl:
add: Dynamic will now play a part in controlling random event antagonists. If a dynamic midround injection is coming too soon, or too early, then the next midround injection will be buffed. Otherwise, it'll spawn normally.
/:cl:

### CODE REVIEW

File: code/game/gamemodes/dynamic/readme.md [hunk 1]
```diff
@@ -168,3 +168,6 @@ The "Dynamic" key has the following configurable values:
 - `threat_curve_width` - A number between 0.5 and 4. Higher value will favour extreme rounds and lower value rounds closer to the average.
 - `roundstart_split_curve_centre` - A number between -5 and +5. Equivalent to threat_curve_centre, but for the budget split. A negative value will weigh towards midround rulesets, and a positive value will weight towards roundstart ones.
 - `roundstart_split_curve_width` - A number between 0.5 and 4. Equivalent to threat_curve_width, but for the budget split. Higher value will favour more variance in splits and lower value rounds closer to the average.
+
+## Random Event "Hijacking"
+TODO
```
Comment (Mothblocks) [inline code/game/gamemodes/dynamic/readme.md:173]: oh yeah oops

**Comment by Timberpoes** [comment]
This has been requested by at least one Headmin informally and is part of the Dynamic 2021 plans.

Issues with Old Dynamic from my own research included midround antags stacked ontop of uncontrollable event spawned antags.

We've put all our eggs in the Dynamic basket. Let's see it through to completion. Pull the trigger on playtesting.

**Comment by itseasytosee** [comment]
The kerning on that png is disgusting.

**Comment by Mothblocks** [comment]
look i know, but draw.io just did that

**Comment by Mothblocks** [comment]
@Timberpoes it works

**Review by Timberpoes — APPROVED** [review]

**Pull request merged**